### PR TITLE
WIP(llgo/rust/opendal): add a opendal demo

### PR DESCRIPTION
**Pull request by spongehah**
Opendal is a data operations middle layer
Now the function and type used in the header file of the c-demo have been migrated, and the official demo has been run in LLGO

### CODE REVIEW

File: rust/opendal/opendal.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	"github.com/goplus/llgo/c"
+	_ "unsafe"
+)
+
+// Write the .pc file for the dylib generated by the Rust library and copy it to pkg-config for proper location.
+const (
+	LLGoPackage = "link: $(pkg-config --libs opendal_c); -lopendal_c"
+)
+
+type Opendal_result_operator_new struct {
```
Comment (luoliwoshang) [inline rust/opendal/opendal.go:13]: Rename `Opendal_result_operator_new` to `ResultOperatorNew` and remove the `Opendal_` prefix from both function and structure names."

**Review by luoliwoshang — CHANGES_REQUESTED** [review]

File: rust/opendal/opendal.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	Op    *Opendal_operator
+	Error *Opendal_error
+}
```
Comment (luoliwoshang) [inline rust/opendal/opendal.go]: Rename `Opendal_result_operator_new` to `ResultOperatorNew` and remove the `Opendal_` prefix from both function and structure names."

**Review by luoliwoshang — CHANGES_REQUESTED** [review]

File: rust/opendal/opendal.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
+package opendal
+
+import (
+	"github.com/goplus/llgo/c"
+	_ "unsafe"
+)
+
+// Write the .pc file for the dylib generated by the Rust library and copy it to pkg-config for proper location.
```
Comment (luoliwoshang) [inline rust/opendal/opendal.go]: Now this comment can be remove

**Review by luoliwoshang — CHANGES_REQUESTED** [review]

File: rust/opendal/opendal.go (hunk 2 above)
Comment (spongehah) [reply rust/opendal/opendal.go]: Already changed

File: rust/opendal/opendal.go [hunk 4]
```diff
@@ -0,0 +1,63 @@
+package opendal
+
+import (
+	"github.com/goplus/llgo/c"
+	_ "unsafe"
+)
```
Comment (luoliwoshang) [inline rust/opendal/opendal.go]:
```suggestion
import (
	_ "unsafe"

	"github.com/goplus/llgo/c"
)
```

**Review by luoliwoshang — CHANGES_REQUESTED** [review]

File: rust/opendal/opendal.go [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+// llgo:link (*Bytes).FreeBytes C.opendal_bytes_free
+func (bs *Bytes) FreeBytes() {
+
+}
+
+// llgo:link (*Operator).FreeOperator C.opendal_operator_free
+func (op *Operator) FreeOperator() {
+
+}
```
Comment (luoliwoshang) [inline rust/opendal/opendal.go]: remove useless line
```suggestion
// llgo:link (*Operator).FreeOperator C.opendal_operator_free
func (op *Operator) FreeOperator() {}
```

**Review by luoliwoshang — CHANGES_REQUESTED** [review]

**Comment by luoliwoshang** [comment]
image
Since `llgo:link` functions are supposed to map directly to certain `C` functions and these mappings may not always require the use of all parameters in the Go context, it might be more appropriate to disable the specific Lint checks for these cases to maintain clean and clear code. Writing `_ = param` in this mapping relationship is not elegant and can clutter the code unnecessarily.
What do you think about this issue @xushiwei  @CarlJi

**Pull request merged**